### PR TITLE
Issue 228. sqlsrv ClientBufferMaxKBSize setting - positive integer.

### PR DESCRIPTION
**Pull request by v-dareck**
In sqlsrv check that ClientBufferMaxKBSize is a positive integer.
Add new tests for sqlsrv and pdo.

### CODE REVIEW

File: source/sqlsrv/util.cpp [hunk 1]
```diff
@@ -592,7 +592,7 @@ PHP_FUNCTION( sqlsrv_configure )
 
             zend_long buffered_query_limit = Z_LVAL_P( value_z );
 
-            CHECK_CUSTOM_ERROR( buffered_query_limit < 0, error_ctx, SQLSRV_ERROR_INVALID_BUFFER_LIMIT, _FN_ ) {
```
Comment (yitam) [inline source/sqlsrv/util.cpp]: Should we update the MSDN doc to specify that buffer size limit must be positive? Just a suggestion.

File: test/pdo_sqlsrv/pdo_228_setAttribute_clientbuffermaxkbsize.phpt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+echo("Client Buffer Size in KB: $attr\n");
+$stmt->execute();	
+$numRows = 0;
+while ($result = $stmt->fetch())
+	$numRows++;
+	
+echo ("Number of rows: $numRows\n");
+
+$size = 3; 
+$stmt = $conn->prepare("SELECT * FROM ".$dbName.".[dbo].[php_test_table_2]", array(constant('PDO::ATTR_CURSOR') => PDO::CURSOR_SCROLL,PDO::SQLSRV_ATTR_CURSOR_SCROLL_TYPE => PDO::SQLSRV_CURSOR_BUFFERED,PDO::SQLSRV_ATTR_CLIENT_BUFFER_MAX_KB_SIZE=> $size));
```
Comment (yitam) [inline test/pdo_sqlsrv/pdo_228_setAttribute_clientbuffermaxkbsize.phpt]: Please explain why the buffer sizes are different for table 1 and table 2

File: test/sqlsrv/srv_228_sqlsrv_clientbuffermaxkbsize.phpt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+// set client buffer size to 0KB returns false
+$ret = sqlsrv_configure('ClientBufferMaxKBSize', 0);
+var_dump($ret);
+// set client buffer size to 1KB
+$size = 1;	
+fetchData($conn, $dbName.".[dbo].[php_test_table_1]", $size); // this should return 0 rows.
+fetchData($conn, $dbName.".[dbo].[php_test_table_2]", $size); // this should return 0 rows.
+// set client buffer size to 2KB
+$size = 2;	
+fetchData($conn, $dbName.".[dbo].[php_test_table_1]", $size); // this should return 1 row.
```
Comment (yitam) [inline test/sqlsrv/srv_228_sqlsrv_clientbuffermaxkbsize.phpt]: Is the client buffer size requirement 2KB due to UTF-8 char set? It's not clear in this test alone

File: test/pdo_sqlsrv/pdo_228_setAttribute_clientbuffermaxkbsize.phpt (hunk 2 above)
Comment (v-dareck) [inline test/pdo_sqlsrv/pdo_228_setAttribute_clientbuffermaxkbsize.phpt]: the data retrieval is different for varchar(max) and varchar(1000).
A slightly bigger buffer is needed for varchar(max) to work even if inserted data is the same.

File: test/sqlsrv/srv_228_sqlsrv_clientbuffermaxkbsize.phpt (hunk 3 above)
Comment (v-dareck) [inline test/sqlsrv/srv_228_sqlsrv_clientbuffermaxkbsize.phpt]: The total column length is > 1024 since it's varchar(1036).

**Comment by Hadis-Fard** [comment]
Please merge this code into 'merged-codebase' branch

**Pull request merged**